docs/dependency_confusion.md

```diff
@@ -143,3 +143,15 @@ registry and will treat the dependency as a plain https dependency. If the url i
 Cachito, it will download the dependency directly without relying on npm/yarn dependency resolution.
 That does not necessarily make using unofficial registries a good idea. If the registry is private,
 your build will either fail or leak internal package names.
+
+## RubyGems
+
+*TL;DR: allowing `https://rubygems.org` as the only source mitigates the issue for Gem dependencies,
+but you still have to be careful about dependencies from Git*
+
+Gemfile.lock pins all dependencies to exact versions. Cachito allows GEM dependencies to be fetched
+only from `https://rubygems.org`, otherwise raises an error. GIT dependencies are specified using 
+a repo URL and a commit hash, so similarly to `gomod`, you are at the mercy of repository 
```

Comment:
The fact that we require a commit hash makes this less problematic than the gomod case.

In golang, dependency versions are tied to git tags -- which can be updated any time. So something like this
```
require github.com/Masterminds/semver v3.1.1
```
points to https://github.com/Masterminds/semver/releases/tag/v3.1.1, which is [7bb0c8](https://github.com/Masterminds/semver/commit/7bb0c843b53d6ad21a3f619cb22c4b442bb3ef3e) right now, but that may change by the time you call `go mod download`.

Gomod protects against this by using the `go.sum` file for verifying checksums, and of course Cachito won't notice the tag changes.

By requiring the exact commit hash, we avoid the problem of mutable git tags

Reply:
Which makes me think: does bundler verify checksums of dependencies? That's the best way to protect against just about any dependency confusion issues

Reply:
Not yet, but Gitlab is porting it's [internal](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/92633) implementation to [Bundler.](https://github.com/rubygems/rubygems/pull/5808), so maybe we'll see this feature in the future.